main.cpp

```diff
@@ -131,287 +172,412 @@ struct CarWash
  This usually means you have to use identical types for all variables used in an expression that is producing that conversion warning.
  */
 
-
-// Thing 1) exercise bike 
 struct ExerciseBike
 {
-// 5 properties:
-//     1) number of seat adjustments (int)
     int numSeatAdjustments = 12;
-//     2) maximum resistance (float)
     float maximumResistance = 15.0f;
-//     3) number of program modes (int)
     int numProgramModes = 8;
-//     4) number of pulse readers (int)
     int numPulseReaders = 2;
-//     5) power consumed (float)
     int powerConsumed = 12.0f;
-// 3 things it can do:
-//     1) begin a workout program
+
     void beginWorkoutProgram(int selectedProgram);
-//     2) read pulse
-    float readPulse(); // returns pulse in beats per minute
-//     3) calculate caloric burn
-    float calculateCaloricBurn(float weight); // returns total calories burned during session
+    float readPulse();
+    float calculateCaloricBurn(float weight, float timeSpent);
 };
 
-// Thing 2) department store
+void ExerciseBike::beginWorkoutProgram(int selectedProgram)
+{
+    std::cout<<"Beginning Workout #"<<selectedProgram<<std::endl;
+}
+
+float ExerciseBike::readPulse()
+{
+    float pulseRead = 30.0f;
+    return pulseRead * 3.0f;
+}
+
+float ExerciseBike::calculateCaloricBurn(float weight, float timeSpent)
+{
+    std::cout<<"Calculating caloric burn..."<<std::endl;
+    return weight * readPulse() * timeSpent;
+}
+
 struct DepartmentStore
 {
-// 5 properties:
-//     1) number of employees (int)
     int numEmployees = 70;
-//     2) number of departments (int)
     int numDepartments = 16;
-//     3) amount of electricity used per month (float)
     float electricityPerWeek = 1200.0f;
-//     4) amount of profit per week (int)
     int profitPerWeek = 60000;
-//     5) name (std::string)
     std::string name = "David's";
-// 3 things it can do:
-//     1) open/close 
-    bool open(); // returns true to denote open store
 
-    bool close(); // returns false to denote closed store
-//     2) host a sale
+    bool open();
+    bool close();
     void hostSale(float discountRate);
-//     3) sell an item to a customer
-    float sellItem(); // returns the total sale price
+    float sellItem(std::string itemName, float price, float taxRate, bool isOnSale, float discountRate);
 };
 
-// Thing 3) golf course
+bool DepartmentStore::open()
+{
+    return 1;
+}
+
+bool DepartmentStore::close()
+{
+    return 0;
+}
+
+void DepartmentStore::hostSale(float discountRate)
+{
+    std::cout<<"We're having a sale! Enjoy "<<discountRate<<"% off!"<<std::endl;
+}
+
+float DepartmentStore::sellItem(std::string itemName, float price, float taxRate, bool isOnSale, float discountRate)
+{
+    float totalPrice = price;
+    if(isOnSale) totalPrice -= (totalPrice * discountRate);
+    totalPrice += (totalPrice * taxRate);
```

Comment:
Put the single-line bodies of `if` blocks on their own line to make the code easier to read.
- [ ] revise